### PR TITLE
Logistic and Triangular Tensors

### PR DESCRIPTION
**Pull request by GeorgeNash**
- Logistic Tensor
- Triangular Tensor

### CODE REVIEW

File: keanu-project/src/main/java/io/improbable/keanu/distributions/tensors/continuous/TensorLogistic.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        final DoubleTensor xMinusAOverB = x.minus(a).divInPlace(b);
+        final DoubleTensor ln1OverB = b.reciprocal().logInPlace();
+
+        return xMinusAOverB.plus(ln1OverB).minusInPlace(
+            xMinusAOverB.expInPlace().plusInPlace(1).logInPlace().timesInPlace(2)
+        );
+    }
+
+    public static Diff dlnPdf(DoubleTensor a, DoubleTensor b, DoubleTensor x) {
+        final DoubleTensor expAOverB = a.div(b).exp();
```
Comment (gordoncaleb) [inline keanu-project/src/main/java/io/improbable/keanu/distributions/tensors/continuous/TensorLogistic.java]: .div(b).expInPlace() ?

File: keanu-project/src/main/java/io/improbable/keanu/distributions/tensors/continuous/TensorLogistic.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        final DoubleTensor expXOverB = x.div(b).exp();
```
Comment (gordoncaleb) [inline keanu-project/src/main/java/io/improbable/keanu/distributions/tensors/continuous/TensorLogistic.java]: div(b).expInPlace()

File: keanu-project/src/main/java/io/improbable/keanu/vertices/dbltensor/SimpleDoubleTensor.java [hunk 3]
```diff
@@ -99,17 +99,17 @@ public DoubleTensor cos() {
 
     @Override
     public DoubleTensor asin() {
-        return this.duplicate().asin();
```
Comment (gordoncaleb) [inline keanu-project/src/main/java/io/improbable/keanu/vertices/dbltensor/SimpleDoubleTensor.java]: Nice catch!

File: keanu-project/src/main/java/io/improbable/keanu/distributions/tensors/continuous/TensorLogistic.java (hunk 1 above)
Comment (GeorgeNash) [reply keanu-project/src/main/java/io/improbable/keanu/distributions/tensors/continuous/TensorLogistic.java]: Nice

File: keanu-project/src/main/java/io/improbable/keanu/distributions/tensors/continuous/TensorLogistic.java (hunk 2 above)
Comment (GeorgeNash) [reply keanu-project/src/main/java/io/improbable/keanu/distributions/tensors/continuous/TensorLogistic.java]: Nice

**Pull request merged**